framework/src/auxkernels/TagMatrixAux.C

```diff
@@ -8,32 +8,32 @@
 //* https://www.gnu.org/licenses/lgpl-2.1.html
 
 #include "TagMatrixAux.h"
+#include "libmesh/utility.h"
 
 registerMooseObject("MooseApp", TagMatrixAux);
 
 InputParameters
 TagMatrixAux::validParams()
 {
-  InputParameters params = AuxKernel::validParams();
+  InputParameters params = TagAuxBase<AuxKernel>::validParams();
   params.addParam<TagName>("matrix_tag", "TagName", "Tag Name this Aux works on");
-  params.addRequiredCoupledVar("v",
-                               "The coupled variable whose components are coupled to AuxVariable");
-  params.set<ExecFlagEnum>("execute_on", true) = {EXEC_TIMESTEP_END};
-
-  params.addClassDescription("Couple the diag of a tag matrix, and return its nodal value");
+  params.addClassDescription("Couple the diagonal of a tag matrix, and return its nodal value");
   return params;
 }
 
 TagMatrixAux::TagMatrixAux(const InputParameters & parameters)
-  : AuxKernel(parameters), _v(coupledMatrixTagValue("v", "matrix_tag"))
+  : TagAuxBase<AuxKernel>(parameters),
+    _v(coupledMatrixTagValue("v", "matrix_tag")),
+    _v_var(*getVar("v", 0))
 {
-  auto & execute_on = getParam<ExecFlagEnum>("execute_on");
-  if (execute_on.size() != 1 || !execute_on.contains(EXEC_TIMESTEP_END))
-    mooseError("execute_on for TagMatrixAux must be set to EXEC_TIMESTEP_END");
+  if (_v_var.feType() != _var.feType())
+    paramError("variable",
+               "The AuxVariable this AuxKernel is acting on has to have the same order and family "
+               "as the variable 'v'");
 }
 
 Real
 TagMatrixAux::computeValue()
 {
-  return _v[_qp];
+  return _scaled ? _v[_qp] : _v[_qp] / _v_var.scalingFactor();
```

Comment:
So the unscaled version of this won't have meaning, either from a system residual/Jacobian standpoint or from a physics forces/fluxes standpoint, if you have nodal boundary conditions feeding into (more accurately ***setting***) the vector/matrix tag because we do not scale nodal boundary condition residuals or Jacobians by the scaling factor.

Reply:
If you have nodal boundary conditions involved with a vector tag, the result will always be 0 (or within tolerance) along the boundary since the nodal boundary condition is the only residual object that determines the residual for those degrees of freedom. Similarly, the Jacobian diagonal associated with those nodes will always just be whatever the nodal boundary condition sets them to be

Reply:
Are you suggesting to remove this change entirely? I think it might still be useful to provide this in case someone wants to investigate the interior of the model.

Reply:
That's fine. I'd just make clear in documentation that this aux kernel could yield surprising results if a nodal boundary condition or other strong constraint adds into the matrix tag.

Reply:
Adding

```
Setting the [!param](/AuxKernels/TagMatrixAux/scaled) parameter to `false` makes
the kernel return values unaffected by variable scaling and automatic scaling.
Note, however, that nodal boundary conditions and strong constraints are
setting Jacobian entries independent of scaling and can lead to unexpected
unscaled results.
```

Reply:
I think that's perfect